### ISSUE TITLE
Allow Script.reload() to not throw errors

### ISSUE THREAD

**Issue by passivestar** [godotengine/godot-proposals#8548]
### Describe the project you are working on

A prototype for an interactive console that lets you execute gdscript both in editor and in game for dev purposes:

https://twitter.com/passivestar_/status/1729992744389468376

<img width="1077" alt="image" src="https://github.com/godotengine/godot-proposals/assets/60579014/6d5db752-7285-4db5-850d-10375257c947">

### Describe the problem or limitation you are having in your project

I'm trying to make it possible to run gdscript code that can either return a value, or not return anything (`void`). When it returns a value, I want to print it out, or use in some UI, etc

This is what I got so far:
```gdscript
static func execute_expression_as_script(expression_string : String) -> void:
	var script := GDScript.new()

	var template = """
@tool
extends EditorScript

static func run():
	_RETURN__CODE_
"""

	# var source_code_with_return = template.replace("_CODE_", expression_string).replace("_RETURN_", "return ")
	var source_code_without_return = template.replace("_CODE_", expression_string).replace("_RETURN_", "")

	# Try with return first
	# script.source_code = source_code_with_return
	# var error := script.reload()

	# if error != OK:
	script.source_code = source_code_without_return
	var error_no_return := script.reload()
	if error_no_return != OK:
		print("Error parsing script")
		return
	script.run()

	# var result = script.run()
	# print(result)
```

As you can see from the commented out code, the idea was to try the version with `return` first, see if it can parse it, and if it can't, try a version without a`return`

That's because if you put something that returns `void` in place of `_CODE_` with a `return` before it, it will result in an error. The error will be printed out to editors output, which isn't great, but what's more problematic is that when you run this in game, it will pause the game to show the error in the debugger. The fact that `Script.reload()` returns an error code doesn't prevent it from throwing an error

In a language with `try/catch` I would just wrap the call to `script.reload()` into a `try`. Since gdscript doesn't have `try/catch`, there's no way to prevent this behaviour

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

If `Script.reload()` accepted a second bool argument to disable errors, it would solve the issue

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When you run `Expression.execute()`, there's an option to suppress errors:
```gdscript
Variant Expression::execute(inputs: Array = [], base_instance: Object = null, show_error: bool = true, const_calls_only: bool = false)
```
Expressions have `has_execute_failed()` and `get_error_text()` that allow you to deal with failed executions however you like

Scripts could do the same:

```gdscript
Error Script::reload(keep_state: bool = false)
```

would become
```gdscript
Error Script::reload(keep_state: bool = false, show_error: bool = true)
```

This wouldn't break anything because it will be `true` by default

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, because the error is thrown by native code

### Is there a reason why this should be core and not an add-on in the asset library?

Can't be fixed with GDScript

**Comment by passivestar** [comment]
> Other way to do this is making "return <void_type>" a valid command that equals to "return" without error.

True, this would solve my particular issue in this case, but perhaps allowing `Script.reload()` to stay silent could solve more potential issues when using `Script` for these things

**Comment by samsface** [comment]
Opening a pr to resolve this. @passivestar try out the build once it's done to see if it's acceptable for your issue: https://github.com/godotengine/godot/pull/85544
